### PR TITLE
Let travis build binaries

### PR DESCRIPTION
**Pull request by danielkucera**
You can use travis to build binaries. The result will look like this:
https://github.com/danielkucera/opendps/releases
All you need to do is add project to travis and set `GITHUB_OAUTH_TOKEN` variable by this guide: https://docs.travis-ci.com/user/deployment/pages/#setting-the-github-token
Release is built with each tag

### CODE REVIEW

**Comment by JackNewman12** [comment]
Beat me to it. This looks good.

There are a few things to consider:
- Should we build multiple models DPS5015 DPS3003? I think it makes sense to check that these build
  - If so, should we also include these images in the release? I think it only makes sense if we have the hardware to test on first.
- The default baudrate should be changed to 9600 in the Makefile since I believe most people will be buying the DPS5005 with the extra $2 USB module (#135)
- This closes ticket #45

**Comment by danielkucera** [comment]
Don't merge it yet, I broke original `build_binaries.sh` script

**Comment by danielkucera** [comment]
But for the other images you just need to edit original `build.sh` to build all needed variants and copy them to `/output`

File: build_binaries.sh [hunk 1]
```diff
@@ -5,26 +5,13 @@
...
-trap cleanup EXIT HUP INT
-
 PROJECT=$(git rev-parse --short HEAD)
 echo "Building opendps git version: $PROJECT"
 
 docker rmi --no-prune opendps_$PROJECT || true
 docker build -t opendps_$PROJECT docker
 
-mkdir output
+mkdir output || true
```
Comment (JackNewman12) [inline build_binaries.sh]: Would `mkdir -p output` be better? That way if there are any errors (besides if the directory already exists) they will show up in the log

**Comment by danielkucera** [comment]
From my side it's ready

**Comment by Xenoamor** [comment]
I think the opendps output files should be renamed to the model that they were built for. Something like opendps_dps5005.bin

**Comment by Xenoamor** [comment]
Looks like the build system expected `BINARY` to match the directory name. Perhaps changing the build script to this? I haven't tested this mind
```
#!/bin/bash

set -eEuo pipefail
umask 022

sudo mkdir -p /home/opendps/code
sudo cp -R /parent/{*,.git*} /home/opendps/code
sudo chown -R opendps:opendps /home/opendps

cd /home/opendps/code
git submodule init
git submodule update
make -j -C libopencm3
make -C opendps fonts "$@"
make -C dpsboot elf bin "$@"

cp -f /home/opendps/code/dpsboot/dpsboot.elf /output/
cp -f /home/opendps/code/dpsboot/dpsboot.bin /output/

make -C opendps elf bin "$@" MODEL=DPS5015
cp -f /home/opendps/code/opendps/opendps.elf /output/opendps_DPS5015.elf
cp -f /home/opendps/code/opendps/opendps.bin /output/opendps_DPS5015.bin

make -C opendps elf bin "$@" MODEL=DPS5005
cp -f /home/opendps/code/opendps/opendps.elf /output/opendps_DPS5005.elf
cp -f /home/opendps/code/opendps/opendps.bin /output/opendps_DPS5005.bin

make -C opendps elf bin "$@" MODEL=DPS3005
cp -f /home/opendps/code/opendps/opendps.elf /output/opendps_DPS3005.elf
cp -f /home/opendps/code/opendps/opendps.bin /output/opendps_DPS3005.bin
```

**Comment by danielkucera** [comment]
I was aiming more towards:
```
make -C opendps elf bin "$@" MODEL=DPS5015
make -C opendps elf bin "$@" MODEL=DPS5005
make -C opendps elf bin "$@" MODEL=DPS3005

cp -f /home/opendps/code/opendps/*.elf /output/
cp -f /home/opendps/code/opendps/*.bin /output/
```
Do you have an idea how to adjust the build system?

**Comment by Xenoamor** [comment]
Change line 52 of libopencm3.rules.mk to: `OBJS		+= opendps.o`

**Comment by Xenoamor** [comment]
The following patch resolves it:
```
diff --git a/dpsboot/Makefile b/dpsboot/Makefile
index e355c81..af31f1f 100644
--- a/dpsboot/Makefile
+++ b/dpsboot/Makefile
@@ -9,6 +9,7 @@ CFLAGS = -I. -I../opendps -DGIT_VERSION=\"$(GIT_VERSION)\" -DCONFIG_PAST_NO_GC -
 #CFLAGS += -flto
 
 OBJS = \
+	dpsboot.o \
 	common.o \
 	hw.o
 
diff --git a/libopencm3.rules.mk b/libopencm3.rules.mk
index 60d6ce8..47e3860 100644
--- a/libopencm3.rules.mk
+++ b/libopencm3.rules.mk
@@ -46,12 +46,6 @@ OPT		:= -Os
 CSTD		?= -std=c99
 
 
-###############################################################################
-# Source files
-
-OBJS		+= $(BINARY).o
-
-
 ifeq ($(strip $(OPENCM3_DIR)),)
 # user has not specified the library path, so we try to detect it
 
diff --git a/opendps/Makefile b/opendps/Makefile
index 02af4da..1a8ba5a 100644
--- a/opendps/Makefile
+++ b/opendps/Makefile
@@ -1,5 +1,3 @@
-BINARY = opendps
-
 # Build with commandline interface rather than serial interface
 COMMANDLINE := 0
 
@@ -24,6 +22,9 @@ WIFI := 1
 # Print debug information on the serial output
 DEBUG ?= 0
 
+# Set the output binary filenames
+BINARY := opendps_$(MODEL)
+
 # Font file
 METER_FONT_FILE ?= gfx/Ubuntu-C.ttf
 METER_FONT_SMALL_SIZE ?= 18
@@ -71,6 +72,7 @@ CFLAGS += \
 LDSCRIPT = stm32f100_app.ld
 
 OBJS = \
+    opendps.o \
     flashlock.o \
     bootcom.o \
     crc16.o \
```

**Comment by danielkucera** [comment]
@Xenoamor I didn't want to touch `libopencm3.rules.mk` so I did it in a different way. Anyway it was a good school of make. Now there is still one problem outstanding - it doesn't rebuild after changing model parameter.

**Comment by Xenoamor** [comment]
I'm almost certain that the `libopencm3.rules.mk` is incorrect as it stands. It should only be used to describe how to build a project and not what sourcefiles are in it.

You'll need to [touch](http://www.linfo.org/touch.html) the `dps-model.h` file to trigger a rebuild I believe

**Comment by danielkucera** [comment]
@Xenoamor yes, touch works. Let's leave the decision to @kanflo whether to edit `libopencm3.rules.mk` or my workaround.
Btw different solution for model config could be using dps-model-$MODEL.c with `const` parameter definitions and then link only the correct file.

**Comment by kanflo** [comment]
As `libopencm3.rules.mk` is a part of this git, please feel free to modify it as needed @danielkucera.

**Comment by danielkucera** [comment]
done

**Comment by Xenoamor** [comment]
Sods law that this has just came out:
https://github.com/features/actions

**Pull request merged**